src/Microsoft.AspNet.WebUtilities/QueryHelpers.cs

```diff
@@ -107,13 +106,21 @@ public static string AddQueryString(string uri, IDictionary<string, string> quer
         /// </summary>
         /// <param name="text">The raw query string value, with or without the leading '?'.</param>
         /// <returns>A collection of parsed keys and values.</returns>
-        public static IDictionary<string, StringValues> ParseQuery(string queryString)
+        public static KeyValueAccumulator ParseQuery(string queryString)
```

Comment:
`Dictionary<string, StringValues> ParseQuery` and `Dictionary<string, StringValues> ParseNullableQuery` like you did for Form?